sympy/sets/sets.py

```diff
@@ -60,7 +60,21 @@ class Set(Basic):
     is_Complement = None
     is_ComplexRegion = False
 
-    is_empty = None
+    @property
+    def is_empty(self):
```

Comment:
The line above would probably do as well.

Reply:
I'm not sure I understand your comment. I replaced the variable `is_empty` by a property method in order to put in the docstring, so that `is_empty` becomes part of the official documentation.

Reply:
I think that it can be a simple attribute here, not a property. Fetching an attribute takes less time than evaluating a property. `'is_empty'` is just a key in `__dict__`. Its implementation does not have to be the same in all subclasses of a class.

Reply:
But how do I add the docstring then? I'm under the impression that it's not possible for simple variables.

I really wouldn't want to replicate the situation as it is for `is_iterable`. There's code that checks for it (I encountered it somewhere in "calculus.py" and wanted to look up some details), but found that it wasn't part of the official SymPy `Set` (class) documentation. I had to grep through the page to find `is_iterable` documented in `ProductSet`!? I had to check the code just to find out if and how `is_iterable` is defined in general and for which set classes it's available. IMO that's not how documentation is supposed to work...

Reply:
I think that it can be documented in the Attributes section of [class docstring](https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring). The difference between a plain data attribute and a property attribute is not visible to the user.

Reply:
We should make a place to document all of the `is_*` attributes/properties. They should probably have their own page in the docs.

Reply:
@jksuom Thanks for the suggestion. I git grepped and had a look at other occurrences of "Attributes" and how it looked in the Sphinx-docs. There are examples in the geometry module. https://docs.sympy.org/latest/modules/geometry/lines.html
Apparently the "Attributes" section is formatted as a table with mostly one short line per attribute, completely separate from other method attributes. That's rather disappointing. For me `is_empty` would fit right in with other properties, such as `is_open`, `is_disjoint`, etc.